crates/primitives/transactions/src/from_broadcasted_transactions.rs

```diff
@@ -113,10 +113,10 @@ impl TryFrom<BroadcastedDeclareTransaction> for UserTransaction {
                     let legacy_contract_class = LegacyContractClass {
                         program: serde_json::from_slice(decompressed_bytes.as_slice())
                             .map_err(|_| BroadcastedTransactionConversionError::ProgramDeserializationFailed)?,
-                        abi: match contract_class.abi.as_ref() {
-                            Some(abi) => Some(abi.iter().cloned().map(|entry| entry.into()).collect()),
-                            None => None,
-                        },
+                        abi: contract_class
+                            .abi
+                            .as_ref()
+                            .map(|abi| abi.iter().cloned().map(|entry| entry.into()).collect()),
```

Comment:
Please replace `entry.into()` with a reference to the associated `from` or `into` function.